sematic/ui/src/pipelines/ScrollingLogView.tsx

```diff
@@ -93,7 +98,42 @@ export default function ScrollingLogView(props: {
     </>);
   }, [pullDownTriggerEnabled, pullDownProgress, isAccumulating, hasMore]);
 
-  const prevIsAccumulating = usePrevious(isAccumulating)
+  const prevIsAccumulating = usePrevious(isAccumulating);
+
+  const renderNextButton = useCallback(() => {
+    return <FooterContainer>
+      {(hasMore && 
+        <Button
+          onClick={accumulateLogsUntilEnd}
+          sx={{ width: "100%" }}
+          disabled={isAccumulating || isLoading}
+          style={{flexShrink: 1}}
+        >
+          {accumulatorButtonMessage}
+        </Button>
+      )}
+      {
+        (!hasMore && !hasReachedBottom && 
+          <Button
+            onClick={scrollToBottom}
+            sx={{ width: "100%" }}
+            style={{flexShrink: 1}}
+          >
+            "Jump to the end"
+          </Button>)
+      }
+    </FooterContainer>;
+  }, [hasMore, hasReachedBottom, isAccumulating, isLoading, 
+    accumulatorButtonMessage, scrollToBottom, accumulateLogsUntilEnd]);
+
+  useEffect(()=> {
+    setFooterRenderProp(renderNextButton);
```

Comment:
Render prop: a callback function to render a section of components dynamically.  Here it delegates a parent to render the footer section.